File: AutomobiliaiPSP1/Lenktynininkas.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AutomobiliaiPSP1
{
    public class Lenktynininkas : IZmogus
    {
        public IVaziavimoStilius Stilius { get; }
        public IAutomobilis TransportoPriemone { get; }
        public Lenktynininkas(IAutomobilis car, IVaziavimoStilius stilius)
        {
            TransportoPriemone = car;
            Stilius = stilius;
        }
        public void Lenktyniauti(int km)
        {
            TransportoPriemone.Vaziuoti(km, Stilius.Daugiklis);
            Console.WriteLine("Pralenktyniavo {0} km", km);
            Console.WriteLine("Liko {0}% bako", TransportoPriemone.LikutisProc());
            Console.WriteLine("Dar galima nuvaziuoti {0}km", TransportoPriemone.LikutisKm());
            Console.WriteLine("");
        }
    }
}

File: Scala/src/Lenktynininkas.scala
abstract class Lenktynininkas {
    def Daugiklis(km: Int): Double
    def Vaziuoti(km: Int, kuroDaugiklis: Double)
    def LikutisProc(): Double
    def LikutisKm(): Int

    def Lenktyniauti(km: Int) {
        Vaziuoti(km, Daugiklis(km))
        Console.println("Pralenktyniavo " + km + "km")
        Console.println("Liko " + LikutisProc() + "% bako")
        Console.println("Dar galima nuvaziuoti " + LikutisKm() + "km")
        Console.println("")
    }
}
